### PR TITLE
Rewrite/encapsulate RealmManager.

### PR DESCRIPTION
**Pull request by Adrian-Weihs**
**Description**
The old RealmsMgr was rewritten into RealmManager and is now in the namespace AscEmu::Realm. The whole logging edited. Every variable is now private. The Method updateRealmPop renamed into setRealmPopulation.

**Todo / Checklist**
- [X] Target is branch develop.
- [X] Detailed description about this pull request.
- [X] Checked AE-Coding standards.

**Tests Performed:** 
- [X] Build AE.
- [X] Server startup.
- [X] Log into world.

### CODE REVIEW

File: src/logonserver/Realm/RealmManager.hpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        Mutex& getServerSocketLock() { return serverSocketLock; }
+        Mutex& getRealmLock() { return realmLock; }
+
+        void timeoutSockets();
+        void checkServers();
+
+        void setRealmOffline(uint32_t realm_id);
+        void setRealmPopulation(uint32_t realm_id, float population);
+
+        void addServerSocket(::LogonCommServerSocket* sock)
```
Comment (Appled) [inline src/logonserver/Realm/RealmManager.hpp:71]: Define functions in cpp files

File: src/logonserver/Realm/RealmManager.hpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        {
+            serverSocketLock.Acquire();
+            serverSockets.insert(sock);
+            serverSocketLock.Release();
+        }
+
+        void removeServerSocket(::LogonCommServerSocket* sock)
```
Comment (Appled) [inline src/logonserver/Realm/RealmManager.hpp:78]: Define functions in cpp files

**Review by Appled — CHANGES_REQUESTED** [review]

File: src/logonserver/Realm/RealmManager.hpp (hunk 1 above)
Comment (Adrian-Weihs) [reply src/logonserver/Realm/RealmManager.hpp]: done

File: src/logonserver/Realm/RealmManager.hpp (hunk 2 above)
Comment (Adrian-Weihs) [reply src/logonserver/Realm/RealmManager.hpp]: done

**Pull request merged**